dao.go

```diff
@@ -30,3 +30,17 @@ func (d *Dao) GetDevice(id int) (*Device, error) {
 	}
 	return nil, nil
 }
+
+func (d *Dao) GetAllDevices() ([]Device, error) {
+	devices := []Device{}
+	for _, dev := range d.data {
+		devices = append(devices, dev)
+	}
+	return devices, nil
+}
+
+func (d *Dao) GetManyDevices(limit int, page int) ([]Device, error) {
```

Comment:
I'm not crazy about this `many` keyword. It's not really very descriptive.

My suggestion: `GetPaginatedDevices`